### PR TITLE
[ROCm] Add stream_executor/stream.h header to pip package

### PR DESCRIPTION
**Pull request by rahulbatra85**
Added stream_executor/stream.h to be included in the pip package

### CODE REVIEW

**Review by rsanthanam-amd — APPROVED** [review]

File: tensorflow/tools/pip_package/BUILD [hunk 1]
```diff
@@ -64,7 +65,9 @@ transitive_hdrs(
         "//tensorflow/python/framework:python_op_gen",
         "//tensorflow/python/client:tf_session_helper",
         "//third_party/eigen3",
-    ] + if_cuda([
+    ] + if_rocm([
```
Comment (rsanthanam-amd) [inline tensorflow/tools/pip_package/BUILD]: Suggestion: maybe we shouldn't gate this for ROCm only.  These forwarding headers would likely benefit CUDA as well.

**Comment by rsanthanam-amd** [comment]
I just saw this:

https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/commit/eba7d976eb072ee27e00d575c3ca9a2c07814b16

So the upstream people had the same idea.

File: tensorflow/tools/pip_package/BUILD (hunk 1 above)
Comment (rahulbatra85) [reply tensorflow/tools/pip_package/BUILD]: My thinking here is that might delay the upstream merge, but I can remove that

**Comment by rahulbatra85** [comment]
> I just saw this:
> 
> [eba7d97](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/commit/eba7d976eb072ee27e00d575c3ca9a2c07814b16)
> 
> So the upstream people had the same idea.

Yes, I saw this too, but the build_pip_package bazel rule don't end up calling that rule at all. I had manually add that to build_rocm_python, but then there was another problem of moving the header file from bazel output to the temporary folder from which wheel is built.

**Comment by rahulbatra85** [comment]
> > I just saw this:
> > [eba7d97](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/commit/eba7d976eb072ee27e00d575c3ca9a2c07814b16)
> > So the upstream people had the same idea.
> 
> Yes, I saw this too, but the build_pip_package bazel rule don't end up calling that rule at all. I had manually add that to build_rocm_python, but then there was another problem of moving the header file from bazel output to the temporary folder from which wheel is built.



A few more details
Here is what happens when we can the build_rocm_python
1. Build the pip-package 
      bazel build --config=opt --config=rocm //tensorflow/tools/pip_package:build_pip_package. This generates bunch of files in bazel-out,    bazel-bin, etc
2. Call the build pip package script. This builds the wheel. It also creates a temporary folder and moves files from bazel-out, bazel-bin that are to be packaged into the wheel. 
   bazel-bin/tensorflow/tools/pip_package/build_pip_package
 
The new rule that upstream ppl added is //tensorflow:install_headers. Two issues with this, first it doesn't get called when //tensorflow/tools/pip_package:build_pip_package is run. Second, it doesn't create entry in the MANIFEST file. So, even though stream_executor/stream.h is present in the bazel-bin, bazel-out paths, it doesn't end up the wheel. With my change, the manifest file is correctly updated and file gets included in the wheel. Also, we don't need to call the new rule at all.

This is the manifest file        
bazel-out/k8-opt/bin/tensorflow/tools/pip_package/build_pip_package.runfiles_manifest

**Comment by rahulbatra85** [comment]
retest Ubuntu-GPU-single please

**Pull request merged**